state/pruner.go

```diff
@@ -59,17 +62,25 @@ func WithPrunerObserver(obs PrunerObserver) PrunerOption {
 	return func(p *prunerConfig) { p.observer = obs }
 }
 
-func NewPruner(stateStore Store, bs BlockStore, logger log.Logger, options ...PrunerOption) *Pruner {
+func NewPruner(
+	stateStore Store,
+	bs BlockStore,
+	indexerService *txindex.IndexerService,
+	logger log.Logger,
+	options ...PrunerOption,
+) *Pruner {
 	cfg := defaultPrunerConfig()
 	for _, opt := range options {
 		opt(cfg)
 	}
 	p := &Pruner{
-		bs:         bs,
-		stateStore: stateStore,
-		logger:     logger,
-		interval:   cfg.interval,
-		observer:   cfg.observer,
+		bs:              bs,
+		stateStore:      stateStore,
+		indexerService:  indexerService,
+		logger:          logger,
+		interval:        cfg.interval,
+		indexerInterval: cfg.indexerInterval,
```

Comment:
For simplicity , I think it is OK to use the same interval for the indexer pruning as for the other pruning jobs.

Reply:
Done